curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/656472ed8f552d2f2b3f7883.md

```diff
@@ -0,0 +1,647 @@
+---
+id: 656472ed8f552d2f2b3f7883
+title: Step 24
+challengeType: 0
+dashedName: step-24
+---
+
+# --description--
+
+Add an `else` block to handle the current song's position in the playlist. 
+
+Within the `else` block, set the `currentTime` property of the `audio` object to the value stored in `userData.songCurrentTime`.
+
+# --hints--
+
+Test 1
+
+```js
+
+```
```

Comment:
The tests for this step:

You should not modify the existing `if` statement and its content.

```js
assert.match(code, /if\s*\(userData\?\.currentSong\s*===\s*null\s*\|\|\s*userData\?\.currentSong\.id\s*!==\s*song\.id\)\s*\{\s*audio\.currentTime\s*=\s*0;?\s*\}/)
```

You should add an `else` block to the existing `if` block.

```js
assert.match(code, /if\s*\(userData\?\.currentSong\s*===\s*null\s*\|\|\s*userData\?\.currentSong\.id\s*!==\s*song\.id\)\s*\{\s*audio\.currentTime\s*=\s*0;?\s*\}\s*else\s{/)
```

You should set `audio.currentTime` to `userData.songCurrentTime` inside the `else` block of your `if` statement.

```js
assert.match(code, /if\s*\(userData\?\.currentSong\s*===\s*null\s*\|\|\s*userData\?\.currentSong\.id\s*!==\s*song\.id\)\s*\{\s*audio\.currentTime\s*=\s*0;?\s*\}\s*else\s{\s*audio\.currentTime\s*=\s*userData\.songCurrentTime;?\s*\}/)
```